backend/appwrite-functions/set-wallet/main.py

```diff
@@ -0,0 +1,39 @@
+import os
+import json
+from appwrite.client import Client
+from appwrite.services.database import Database
+
+# provided by default
+USER_ID = os.environ.get('APPWRITE_FUNCTION_USER_ID')
+PAYLOAD = os.environ.get('APPWRITE_FUNCTION_DATA')
+APPWRITE_PROJECT = os.environ.get('APPWRITE_FUNCTION_PROJECT_ID')
+
+# provided by cloud function env vars
+APPWRITE_ENDPOINT = os.environ.get('APPWRITE_ENDPOINT')
+APPWRITE_API_KEY = os.environ.get('APPWRITE_API_KEY')
+WALLETS_COLLECTION_ID = os.environ.get('WALLETS_COLLECTION_ID')
+
+client = Client()
+database = Database(client)
+
+(client
+ .set_endpoint(APPWRITE_ENDPOINT)
+ .set_project(APPWRITE_PROJECT)
+ .set_key(APPWRITE_API_KEY)
+ )
+
+desiredWalletAddress = json.loads(PAYLOAD)["walletAddress"]
+
+readResult = database.list_documents(WALLETS_COLLECTION_ID, ["userId=" + USER_ID])["documents"]
+
+if len(readResult) > 0:
+    writeResult = database.update_document(WALLETS_COLLECTION_ID, readResult[0]["$id"],
+                                           {"walletAddress": desiredWalletAddress})
+else:
+    writeResult = database.create_document(WALLETS_COLLECTION_ID, {"userId": USER_ID, "walletAddress": desiredWalletAddress})
```

Comment:
I think we should also allow admins to read this document, no?

Reply:
AFAIK admins are also users

Reply:
But they also belong to the admin team afaik (based on the other scripts where we add some admins). To the best of my knowledge, if you only set one user, then only that user can read it

Reply:
Why should admins be able to see any users address?

Reply:
I have no use case for it now, but shouldn't admins have access to general user data? If you do not think there is a use case for it, leave admins out

Reply:
As long as we don't have a use case for that i would restrict access as far as possible